arcgis-ios-sdk-samples/Scenes/Animate 3D graphic/CameraSettingsViewController.swift

```diff
@@ -16,141 +16,120 @@
-        self.setInitialValues()
+        guard let cameraController = orbitGeoElementCameraController else {
+            return
+        }
         
-        //add observers to update the sliders
-        self.orbitGeoElementCameraController?.addObserver(self, forKeyPath: #keyPath(AGSOrbitGeoElementCameraController.cameraDistance), options: .new, context: nil)
-        self.orbitGeoElementCameraController?.addObserver(self, forKeyPath: #keyPath(AGSOrbitGeoElementCameraController.cameraHeadingOffset), options: .new, context: nil)
-        self.orbitGeoElementCameraController?.addObserver(self, forKeyPath: #keyPath(AGSOrbitGeoElementCameraController.cameraPitchOffset), options: .new, context: nil)
+        // apply initial values to controls
+        updateUIForDistance()
+        updateUIForHeadingOffset()
+        updateUIForPitchOffset()
+        autoHeadingEnabledSwitch.isOn = cameraController.isAutoHeadingEnabled
+        autoPitchEnabledSwitch.isOn = cameraController.isAutoPitchEnabled
+        autoRollEnabledSwitch.isOn = cameraController.isAutoRollEnabled
+        
+        // add observers to the values we want to show in the UI
+        distanceObservation = cameraController.observe(\.cameraDistance) {[weak self] (controller, change) in
+            self?.updateUIForDistance()
+        }
+        headingObservation = cameraController.observe(\.cameraHeadingOffset) {[weak self] (controller, change) in
+            self?.updateUIForHeadingOffset()
+        }
+        pitchObservation = cameraController.observe(\.cameraPitchOffset) {[weak self] (controller, change) in
+            self?.updateUIForPitchOffset()
+        }
     }
     
-    private func setInitialValues() {
-        
-        guard let cameraController = self.orbitGeoElementCameraController else {
+    private let numberFormatter = NumberFormatter()
+    
+    private func updateUIForDistance() {
+        guard let cameraController = orbitGeoElementCameraController else {
             return
         }
-        
-        self.distanceSlider.value = Float(cameraController.cameraDistance)
-        self.distanceLabel.text = "\(Int(cameraController.cameraDistance))"
-        
-        self.headingOffsetSlider.value = Float(cameraController.cameraHeadingOffset)
-        self.headingOffsetLabel.text = "\(Int(cameraController.cameraHeadingOffset))º"
-        
-        self.pitchOffsetSlider.value = Float(cameraController.cameraPitchOffset)
-        self.pitchOffsetLabel.text = "\(Int(cameraController.cameraPitchOffset))º"
-        
-        self.autoHeadingEnabledSwitch.isOn = cameraController.isAutoHeadingEnabled
-        self.autoPitchEnabledSwitch.isOn = cameraController.isAutoPitchEnabled
-        self.autoRollEnabledSwitch.isOn = cameraController.isAutoRollEnabled
+        distanceSlider.value = Float(cameraController.cameraDistance)
+        distanceLabel.text = numberFormatter.string(from: cameraController.cameraDistance as NSNumber)! + " m"
```

Comment:
It would be good to use a measurement formatter in this view controller as well.